src/libtiled/maprenderer.cpp

```diff
@@ -164,8 +165,15 @@ void CellRenderer::render(const Cell &cell, const QPointF &pos, const QSizeF &si
     if (origin == BottomCenter)
         fragment.x -= sizeHalf.x();
 
-    if (cell.flippedAntiDiagonally()) {
-        fragment.rotation = 90;
+    qreal rotation = 0;
+    if (mIsHexagonal) {
+        rotation = (cell.flippedLeftHexAntiDiagonally() ? 300 : (cell.flippedAntiDiagonally() ? 60 : 0));
+    } else {
+        rotation = (cell.flippedAntiDiagonally() ? 90 : 0);
+    }
+
+    if (rotation != 0) {
```

Comment:
Here, please keep the `if (cell.flippedAntiDiagonally()) {` check and simplify the hexagonal case first by having something like:
```c++
if (mCellType == HexagonalCells) {
    if (cell.rotated60())
        fragment.rotation += 60;
    if (cell.rotated120())
        fragment.rotation += 120;
} else if (cell.flippedAntiDiagonally()) {
    ...
}
```